### PR TITLE
Cleanup dask.array.reductions

### PR DESCRIPTION
**Pull request by jcrist**
Improve dtype compatibility with numpy, increase testing of functions.

### CODE REVIEW

**Comment by jcrist** [comment]
I'm not sure what the error is caused by - for me this failure happens on current master, and has nothing to do with the changes I made.

**Comment by mrocklin** [comment]
I've been bad and pushing to master a bit lately.  This might be my fault.  Will check it out.

**Comment by jcrist** [comment]
It's due to a new version of pandas being released - we're missing a "name" for the series in the result that pandas now has. Looking into why this happens now.

File: dask/array/core.py [hunk 1]
```diff
@@ -769,7 +769,7 @@ def topk(self, k):
 
     def astype(self, dtype, **kwargs):
         """ Copy of the array, cast to a specified type """
-        return elemwise(partial(np.ndarray.astype, dtype=dtype, **kwargs),
+        return elemwise(partial(lambda x, dtype, **kwargs: x.astype(dtype, **kwargs), dtype=dtype, **kwargs),
```
Comment (mrocklin) [inline dask/array/core.py]: It seems odd to have a lambda within a partial.  Can these be fused to either a lambda or a partial

Comment (jcrist) [inline dask/array/core.py]: Woops, forgot to clean this up. Will do.

**Comment by mrocklin** [comment]
This seems fine to me.  OK to merge?

**Pull request merged**